openedx/core/djangoapps/user_api/management/tests/test_email_opt_in_list.py

```diff
@@ -257,6 +258,24 @@ def _cleanup():  # pylint: disable=missing-docstring
         with self.assertRaisesRegexp(CommandError, "^File already exists"):
             email_opt_in_list.Command().handle(temp_file.name, self.TEST_ORG)
 
+    def test_no_user_profile(self):
+        """
+        Tests that command does not break if a user has no profile.
+        """
+        self._create_courses_and_enrollments((self.TEST_ORG, True))
+        self._set_opt_in_pref(self.user, self.TEST_ORG, True)
+
+        # Remove the user profile, and re-fetch user
+        self.assertTrue(hasattr(self.user, 'profile'))
+        self.user.profile.delete()
+        user = User.objects.get(id=self.user.id)
```

Comment:
@aamir-khan `self.user.profile` is not updated after deleting the profile using `self.user.profile.delete()` so I happened to fetch the user again.